e2e/libs/keywords/volume_keywords.py

```diff
@@ -1,73 +1,310 @@
 
-    def delete_replica(self, volume_name, replica_index):
-        replica_node_name = Node().get_by_index(int(replica_index))
-        print(f"delete volume {volume_name}'s replica\
-                {replica_index} {replica_node_name}")
-        self.volume.delete_replica(volume_name, replica_node_name)
+        logging.info(
+            f"deleting volume {volume_name}'s replica on the node {node_name}")
+        self.volume.delete_replica(volume_name, node_name)
 
+    def wait_for_replica_rebuilding_start(self, volume_name, node_index):
+        node_name = Nodes.get_name_by_index(int(node_index))
+        if node_name == "":
+            raise Exception(f"failed to get node name with index {node_index}")
 
-    def wait_for_replica_rebuilding_start(self, volume_name, replica_index):
-        replica_node_name = Node().get_by_index(int(replica_index))
-        print(f"wait for  volume {volume_name}'s replica\
-                {replica_index} {replica_node_name} rebuilding started")
-        self.volume.wait_for_replica_rebuilding_start(
-            volume_name,
-            replica_node_name
-        )
+        logging.info(
+            f"waiting the {volume_name} replica on node {node_name} rebuilding start")
+        self.volume.wait_for_replica_rebuilding_start(volume_name, node_name)
 
+    def wait_for_replica_rebuilding_complete(self, volume_name, node_index):
+        node_name = Nodes.get_name_by_index(int(node_index))
+        if node_name == "":
+            raise Exception(f"failed to get node name with index {node_index}")
 
-    def wait_for_replica_rebuilding_complete(self, volume_name, replica_index):
-        replica_node_name = Node().get_by_index(int(replica_index))
-        print(f"wait for  volume {volume_name}'s replica\
-                {replica_index} {replica_node_name} rebuilding completed")
+        logging.info(
+            f"waiting the {volume_name} replica on node {node_name} rebuilding complete")
         self.volume.wait_for_replica_rebuilding_complete(
-            volume_name,
-            replica_node_name
-        )
-
+            volume_name, node_name)
 
     def cleanup_resources(self):
-        print('cleanup_resources')
+        logging.info('cleaning up resources')
         self.node_exec.cleanup()
+        self.node_operation.cleanup()
+        self.volume.clean_up()
+
+    def cleanup_resources_on_node(self):
+        self.node_exec.cleanup()
+
+    def restore_node_state(self):
+        self.node_operation.cleanup()
+
+    def cleanup_longhorn_resources(self):
+        self.volume.clean_up()
+
+    def power_off_node(self, node_index):
+        node_name = Nodes.get_name_by_index(int(node_index))
+        if node_name == "":
+            raise Exception(f"failed to get node name with index {node_index}")
+
+        logging.info(f"powering off the node {node_name}")
+        self.node_operation.power_off_node(node_name=node_name)
+
+    def power_on_node(self, node_index):
```

Comment:
I moved the keyword functions from `volume_keywords.py` to the respective python files for the workload keywords to avoid volume-unrelated functions in the same file.

├── keywords: Keyword definitions for workloads
        ├── __init__.py
        ├── common_keywords.py
        ├── engine_keywords.py
        ├── node_keywords.py
        ├── pod_keywords.py
        ├── replica_keywords.py
        └── volume_keywords.py